spec/features/state_file/login_spec.rb

```diff
@@ -92,6 +92,171 @@
+                to: phone_number,
+                body: "Your 6-digit FileYourStateTaxes verification code is: #{verification_code}. This code will expire after 10 minutes.",
+                )
+
+              expect(page).to have_text "Enter the code to continue"
+              fill_in "Enter the 6-digit code", with: verification_code
+              click_on "Verify code"
+
+              expect(page).to have_text I18n.t("state_file.intake_logins.edit.title")
+              fill_in "Enter your Social Security number or ITIN. For example, 123-45-6789.", with: ssn
+              click_on "Continue"
+
+              expect(page).to have_text "let me edit the response XML"
+            end
+          end
+
+          context "gets SSN wrong" do
+            let(:wrong_ssn) { "887223344" }
+
+            before do
+              allow(SsnHashingService).to receive(:hash).with(wrong_ssn).and_return "wrong_hashed_ssn"
+            end
+
+            scenario "will be able to re-enter SSN 3 times" do
+              visit "/login-options"
+              expect(page).to have_text "Sign in to FileYourStateTaxes"
+              click_on "Sign in with phone number"
+
+              expect(page).to have_text "Sign in with your phone number"
+              fill_in "Your phone number", with: phone_number
+              perform_enqueued_jobs do
+                click_on I18n.t("state_file.questions.email_address.edit.action")
+              end
+
+              expect(twilio_service).to have_received(:send_text_message).with(
+                to: phone_number,
+                body: "Your 6-digit FileYourStateTaxes verification code is: #{verification_code}. This code will expire after 10 minutes.",
+                )
+
+              expect(page).to have_text "Enter the code to continue"
+              fill_in "Enter the 6-digit code", with: verification_code
+              click_on "Verify code"
+
+              expect(page).to have_text I18n.t("state_file.intake_logins.edit.title")
+              fill_in "Enter your Social Security number or ITIN. For example, 123-45-6789.", with: wrong_ssn
+              click_on "Continue"
+
+              expect(page).to have_text I18n.t("state_file.intake_logins.edit.title")
+              expect(page).to have_text I18n.t("state_file.intake_logins.form.errors.bad_input")
+              fill_in "Enter your Social Security number or ITIN. For example, 123-45-6789.", with: wrong_ssn
+              click_on "Continue"
+
+              expect(page).to have_text I18n.t("state_file.intake_logins.edit.title")
+              expect(page).to have_text I18n.t("state_file.intake_logins.form.errors.bad_input")
+              fill_in "Enter your Social Security number or ITIN. For example, 123-45-6789.", with: wrong_ssn
+              click_on "Continue"
+
+              expect(page).to have_text I18n.t("state_file.intake_logins.account_locked.title")
+            end
+          end
+        end
+      end
+    end
```

Comment:
is having feature specs for this helpful? I guess it's really a repeat of specs in the `intake_logins_controller_spec`, and hesitant to add to the fyst flow since it's sort of the same thing as what's in the `verification_code_controller` but happy to add one for that side as well.

Reply:
🤔 feature tests add to build time, so when things can be tested in the controller, i feel okay about not including them. but i will leave it up to you!